File: arrays_and_hashing/longest_consecutive_sequence.py
# Memory: 8 MB•Time: 27ms•Submitted at: 04/20/2026 12:57
# Beats 100% in runtime, 56.34% in nmemory

# Given an array of integers nums, return the length of the longest consecutive sequence of elements that can be formed.

# A consecutive sequence is a sequence of elements in which each element is exactly 1 greater than the previous element. The elements do not have to be consecutive in the original array.

# You must write an algorithm that runs in O(n) time.

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        result = 0
        tmp_result = 0

        # Empty list case
        if (len(nums) == 0):
            return 0

        values = set(nums)

        for num in values:
            if num - 1 not in values: 
                # Start of sequence
                curr_num = num
                tmp_result = 0

                # Check sequence length
                while curr_num + 1 in values:
                    tmp_result += 1
                    curr_num += 1
                    result = max(tmp_result, result)

        return result + 1

File: arrays_and_hashing/longest_consecutive_sequence.cpp
// Memory: 106.5 MB•Time: 2ms•Submitted at: 04/20/2026 11:35
// Beats 99.7% in runtime, 94.89% in memory

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int result = 0;
        int tmp_result = 0;

        // Empty vector case
        if (nums.size() == 0) {
            return 0;
        }

        // Sort the array
        sort(nums.begin(), nums.end());

        // Remove duplicates
        nums.erase(unique(nums.begin(), nums.end()), nums.end());

        // Count consecutive terms
        for(int i = 0; i < nums.size(); i++) {
            if (nums[i+1] == nums[i] + 1) {
                tmp_result ++;
                if (tmp_result > result) {
                    result = tmp_result;
                }
            } else {
                tmp_result = 0;
            }
        }

        return result + 1;
    }
};
